plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryTypeMapping.java

```diff
@@ -562,10 +562,28 @@ public void testTimestampWithTimeZone()
                 // max value in BigQuery
                 .addRoundTrip("TIMESTAMP", "TIMESTAMP '9999-12-31 23:59:59.999999-00:00'",
                         TIMESTAMP_TZ_MICROS, "TIMESTAMP '9999-12-31 23:59:59.999999 UTC'")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test.timestamp_tz"))
                 .execute(getQueryRunner(), bigqueryCreateAndInsert("test.timestamp_tz"));
         // TODO (https://github.com/trinodb/trino/pull/12210) Add support for timestamp with time zone type in views
     }
 
+    @Test
+    public void testUnsupportedTimestampWithTimeZone()
+    {
+        try (TestTable table = new TestTable(getBigQuerySqlExecutor(), "test.unsupported_tz", "(col timestamp)")) {
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES (timestamp '-0001-01-01 00:00:00.000000 UTC')", "Failed to insert rows.*");
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES (timestamp '0000-12-31 23:59:59.999999 UTC')", "Failed to insert rows.*");
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES (timestamp '10000-01-01 00:00:00.000000 UTC')", "Failed to insert rows.*");
```

Comment:
Please add another case of valid timestamp with negative year. e.g. -2021-09-07 23:59:59.999999 UTC
The current logic doesn't throw an exception even though it's out of supported range.